### ISSUE TITLE
Add documention to the project

### ISSUE THREAD

**Issue by PiotrekDlang** [JinShil/utiliD#11]
Hello Mike,

I would like to add some documentation to the project
In the initial form I plan to gather all responses from you and key D users.

Please say your opinion on this. Also if you have some ideas or ongoing work, please let me know.

**Comment by JinShil** [comment]
I'm not sure what kind of documentation you had in mind, so I'll just express my general thoughts for how:

First, and foremost, I want to make any documentation easy to maintain.  For example in a DMD PR a while ago, I deleted a backend.txt file that had each file listed with a description.  And then each module in the backend also had a description at the top of the module.  I don't want to have to maintain that redundancy manually.  I'd prefer that the module description should be added top of each module in the form of a DDoc comment, and the documentation generated with D's documentation generation facilities.

For code documentation, at a minimum, each public interface should have a DDoc comment.  I think we can start using Markdown with the -preview=markdown feature now, and I would prefer Markdown over DDoc.  Then I'd like to have the API documentaiton generated automatically and published at GitHub pages or something like that so users can read the API without being distracted by the code.

The repository README file should desribe:
*  What this repository is and why it exists
*  How people can use it
  * How to build the library and how to use it in their project
*  How people can contribute to it (i.e. specific tasks that need to be done)
  * How to run the test suite locally

Other README files may exist in the repository to help contributors understand the structure of the code, test suite, etc to make it easier for them to understand the implementation, and guide them so they know where a specific feature should exists, and to ensure users can import in a pay-as-you-go fashion.

That's all I have for now. Please clarify if my response above is lacking in any way.

**Comment by PiotrekDlang** [comment]
Yeah, I was initial thinking about what you wrote in the last part, i.e goals, design and ideas. 
I thought it would be nice to have all related information in one place. Something like you wrote in #10.
